generator/uscore/metadata_extractor.rb

```diff
@@ -226,6 +228,47 @@ def add_search_param_descriptions(profile_definition, sequence)
         end
       end
 
+      def add_valid_codes(profile_definition, profile_element, fhir_metadata, param_metadata, path)
+        if param_metadata[:contains_multiple]
+          add_values_from_slices(param_metadata, profile_definition, path)
+        elsif param_metadata[:type] == 'CodeableConcept'
+          add_values_from_fixed_codes(param_metadata, profile_definition, profile_element)
+          add_values_from_patterncodeableconcept(param_metadata, profile_element)
+        end
+        add_values_from_valueset_binding(param_metadata, profile_element)
+        add_values_from_resource_metadata(param_metadata, fhir_metadata)
```

Comment:
if the binding on the resource is `example`, but the binding on the defining profile is `required` it looks like the resource values will still be added (even though they probably shouldn't be).

Reply:
It'll only add values from the resource if it didn't find any from the profile (checks in add_values_from_resource_metadata) so I don't think we need to worry about this.

Reply:
ahhhh I see. based on the method name I expected it was just adding them.  I do think it would be nice to have the embedded if statement up in this method to keep them all together, but the method is so small I'm cool leaving it for now.